src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/FileIO/TextFieldParser.vb

```diff
@@ -0,0 +1,1712 @@
+        Private m_Buffer(DEFAULT_BUFFER_LENGTH - 1) As Char
+
+        ' The minimum length for a valid fixed width line
+        Private m_LineLength As Integer
+
+        ' Indicates whether or not we handle quotes in a csv appropriate way
+        Private m_HasFieldsEnclosedInQuotes As Boolean = True
+
+        ' A string of the chars that count as spaces (used for csv format). The norm is spaces and tabs.
+        Private m_SpaceChars As String
+
+        ' The largest size a line can be.
+        Private m_MaxLineSize As Integer = 10000000
+
+        ' The largest size the buffer can be
+        Private m_MaxBufferSize As Integer = 10000000
+
+        ' Regex pattern to determine if field begins with quotes
+        Private Const BEGINS_WITH_QUOTE As String = "\G[{0}]*"""
+
+        ' Regex pattern to find a quote before a delimiter
+        Private Const ENDING_QUOTE As String = """[{0}]*"
+
+        ' Indicates passed in stream should be not be closed
+        Private m_LeaveOpen As Boolean = False
+    End Class
+
+    ''' <summary>
+    '''  Enum used to indicate the kind of file being read, either delimited or fixed length
+    ''' </summary>
+    ''' <remarks></remarks>
+    Public Enum FieldType As Integer
+        '!!!!!!!!!! Changes to this enum must be reflected in ValidateFieldTypeEnumValue()
+        Delimited
+        FixedWidth
+    End Enum
+
+    ''' <summary>
+    '''  Helper class that when passed a line and an index to a quote delimited field
+    '''  will build the field and handle escaped quotes
+    ''' </summary>
+    ''' <remarks></remarks>
+    Friend Class QuoteDelimitedFieldBuilder
+        ''' <summary>
+        '''  Creates an instance of the class and sets some properties
+        ''' </summary>
+        ''' <param name="DelimiterRegex">The regex used to find any of the delimiters</param>
+        ''' <param name="SpaceChars">Characters treated as space (usually space and tab)</param>
+        Public Sub New(ByVal DelimiterRegex As Regex, ByVal SpaceChars As String)
+            m_DelimiterRegex = DelimiterRegex
+            m_SpaceChars = SpaceChars
+        End Sub
+
+        ''' <summary>
+        '''  Indicates whether or not the field has been built.
+        ''' </summary>
+        ''' <value>True if the field has been built, otherwise False</value>
+        ''' <remarks>If the Field has been built, the Field property will return the entire field</remarks>
+        Public ReadOnly Property FieldFinished() As Boolean
+            Get
+                Return m_FieldFinished
+            End Get
+        End Property
+
+        ''' <summary>
+        '''  The field being built
+        ''' </summary>
+        ''' <value>The field</value>
+        ''' <remarks></remarks>
+        Public ReadOnly Property Field() As String
+            Get
+                Return m_Field.ToString()
+            End Get
+        End Property
+
+        ''' <summary>
+        '''  The cu
```

Comment:
Please inline this constant, to match the reference source. Sorry, I know I asked you to move this change here previously but I think it makes sense to simply revert the change to minimize the differences. Thanks.